webapp/src/lib/components/spot-component/spot-location-create-form.svelte

```diff
@@ -5,100 +5,280 @@
         Form,
         Select,
         SelectItem,
-        TextInput
+        TextInput,
+        Search,
+        Modal
     } from 'carbon-components-svelte';
+    import type { AddressResult } from '$lib/types/address/address';
+
+    const apiKey = import.meta.env.VITE_GEOCODING_API_KEY;
+
+    let searchQuery = '';
+    let debounceTimer: number | undefined;
+    let results: AddressResult[];
+    let dropdownOpen: boolean = false;
+    let warningModalOpen: boolean = false;
+    let warningText: string = '';
 
     export let isReadOnly: boolean;
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    export let handleSubmit = (_: Event) => {};
+    export let isVerified: boolean;
 
+    export let houseNumber: number;
     export let streetAddress: string;
     export let city: string;
     export let state: string;
+    export let country: string;
     export let countryCode: string;
     export let postalCode: string;
     export let hasShelter: boolean;
     export let hasPlugIn: boolean;
     export let hasChargingStation: boolean;
 
+    // Province data mapping
     const provinceData: Map<string, string> = new Map([
         ['MB', 'Manitoba'],
         ['ON', 'Ontario'],
         ['AB', 'Alberta'],
         ['QC', 'Quebec'],
         ['NS', 'Nova Scotia'],
         ['BC', 'British Columbia'],
-        ['NL', 'Newfouundland and Labrador'],
+        ['NL', 'Newfoundland and Labrador'],
         ['PE', 'Prince Edward Island'],
         ['SK', 'Saskatchewan'],
         ['YT', 'Yukon'],
         ['NU', 'Nunavut'],
         ['NT', 'Northwest Territories']
     ]);
+
+    // Function to fetch address suggestions
+    const searchAddress = () => {
+        const url = `https://api.geoapify.com/v1/geocode/search?text=${encodeURIComponent(searchQuery)}&apiKey=${apiKey}`;
+
+        fetch(url)
+            .then((response) => response.json())
+            .then((data) => {
+                results = data.features ?? [];
+                dropdownOpen = true;
```

Comment:
How about moving the search and verify outside of the component to +page.svelte? The component doesn't need to know about any of this logic in my opinion, It should only care about displaying stuff.